kratos/geometries/geometry.h

```diff
@@ -389,6 +385,218 @@ class Geometry : public PointerVector<TPointType>
         return *this;
     }
 
+     operator PointsArrayType&()
+    {
+        return *mpPointVector;
+    }
+
+    ///@}
+    ///@name PointerVector Operators
+    ///@{
+
+    TPointType& operator[](const SizeType& i)
+    {
+        return (*mpPointVector)[i];
+    }
+
+    TPointType const& operator[](const SizeType& i) const
+    {
+        return (*mpPointVector)[i];
+    }
+
+    PointPointerType& operator()(const SizeType& i)
+    {
+        return (*mpPointVector)(i);
+    }
+
+    ConstPointPointerType& operator()(const SizeType& i) const
+    {
+        return (*mpPointVector)(i);
+    }
+
+    //bool operator==(const Geometry& r) const // nothrow
+    //{
+    //    if (size() != r.size())
+    //        return false;
+    //    else
+    //        return std::equal(
+    //            (*mpPointVector).begin(),
+    //            (*mpPointVector).end(),
+    //            (*r.mpPointVector).begin(),
+    //            this->EqualKeyTo());
+    //}
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    iterator                   begin()
```

Comment:
@pooyan-dadvand how many of the operations that are copied from the PointerVector should remain here?
according to @tteschemacher many of them are used somewhere but I am not sure if `Geometry` should have e.g. a `ptr_rend` method

Maybe we could deprecate the ones that we decide to drop?